### DISCUSSION TITLE
How to clean up pre-flashed devices after purchase

### DISCUSSION THREAD

**Discussion by Th0maz** [arendst/Tasmota#18669] [Q&A]
Hi,
I bought a pre-flashed device ([RocketController ASTRA](https://templates.blakadder.com/rocket_R4A4.html)) with Tasmota already on it.
How can I make sure that there's a "clean" tasmota software on this device without any unwanted components (built into firmware or configured somehow)?
Should I do an OTA re-flash followed by a factory reset?
Thanks for your input :)
Thomas

**Comment by sfromis** [comment]
> Should I do an OTA re-flash followed by a factory reset?

That is generally a reasonable plan.

However, when it comes to ESP32, that would not make sure to establish the standard Tasmota partition scheme (with a large app partition, a small file system partition, and a safeboot partition to have OTA updates over half the flash size. Depending on existing partition layout, you may be able to get that in place by running the Partition Wizard tool. If serial flashing is easy to do, that may be the more straightforward approach.

To be clear, not having any experience with that device, I do not know about what partition scheme it comes with, or if there is actually any issue about the partitioning.

**Comment by Th0maz** [comment]
I booted the device up for the first time. Is this the default partition layout of the ESP32 tasmota image?

![grafik](https://github.com/arendst/Tasmota/assets/12065628/457c804c-d699-4ed8-b7ce-714d87f3c29b)

![grafik](https://github.com/arendst/Tasmota/assets/12065628/1184c028-d155-4ec6-9c1c-d020ce1c8b58)

**Reply by sfromis** [reply]
Two same-sized partitions of 1856 KB, along with a 320 KB file system partition indeed looks like the old default layout for Tasmota on ESP32. While it can be upgraded to the safeboot layout, this is not a requirement for running current builds from the Tasmota project, as long as they fit within the 1856 KB partition size.

**Comment by Th0maz** [comment]
Thanks for the clarification.
As it was pretty easy to open the device and do a serial flash (with previous delete), I chose to do this.
Device is up and running again and I feel better about having a trustworthy firmware version on it :)
